tm2py/model/assignment/skim_transit_network.py

```diff
@@ -0,0 +1,1565 @@
+           +(min_stand_weight+(max_stand_weight-min_stand_weight)*(transit_volume/capacity)**power_stand_weight)*num_standing
+           )/(transit_volume+0.01)))
+
+    # Toronto implementation limited factor between 1.0 and 10.0
+    return crowded_factor
+"""
+
+_headway_cost_function = """
+max_hdwy_growth = 1.5
+max_hdwy = 999.98
+
+
+def calc_eawt(segment, vcr, headway):
+    # EAWT_AM = 0. 259625 + 1. 612019*(1/Headway) + 0.005274*(Arriving V/C) + 0. 591765*(Total Offs Share)
+    # EAWT_MD = 0. 24223 + 3.40621* (1/Headway) + 0.02709*(Arriving V/C) + 0. 82747 *(Total Offs Share)
+    line = segment.line
+    prev_segment = line.segment(segment.number - 1)
+    alightings = 0
+    total_offs = 0
+    all_segs = iter(line.segments(True))
+    prev_seg = next(all_segs)
+    for seg in all_segs:
+        total_offs += prev_seg.transit_volume - seg.transit_volume + seg.transit_boardings
+        if seg == segment:
+            alightings = total_offs
+        prev_seg = seg
+    if total_offs < 0.001:
+        total_offs = 9999  # added due to divide by zero error
+    if headway < .01:
+        headway = 9999
+    eawt = 0.259625 + 1.612019*(1/headway) + 0.005274*(vcr) + 0.591765*(alightings / total_offs)
+    # if mode is LRT / BRT mult eawt * 0.4, if HRT /commuter mult by 0.2
+    # use either .mode.id or ["#src_mode"] if fares are used
+    mode_char = line{0}
+    if mode_char in ["l", "x"]:
+        eawt_factor = 0.4
+    elif mode_char in ["h", "c", "f"]:
+        eawt_factor = 0.2
+    else:
+        eawt_factor = 1
+    return eawt * eawt_factor
+
+
+def calc_adj_headway(transit_volume, transit_boardings, headway, capacity, segment):
+    prev_hdwy = segment["@phdwy"]
+    delta_cap = max(capacity - transit_volume + transit_boardings, 0)
+    adj_hdwy = min(max_hdwy, prev_hdwy * min((transit_boardings+1) / (delta_cap+1), 1.5))
+    adj_hdwy = max(headway, adj_hdwy)
+    return adj_hdwy
+
+def calc_headway(transit_volume, transit_boardings, headway, capacity, segment):
+    vcr = transit_volume / capacity
+    eawt = calc_eawt(segment, vcr, segment.line.headway)
+    adj_hdwy = calc_adj_headway(transit_volume, transit_boardings, headway, capacity, segment)
+    return adj_hdwy + eawt
+
+"""
+
+
+def connect_to_desktop(port=59673):
+    desktop = _app.connect(port=port)
+    return desktop
+
+
+def start_desktop(root, title="mtc_emme", port=59673):
+    emme_project = _os.path.join(root, title, title + ".emp")
+    desktop = _app.start(  # will not close desktop when program ends
+        project=emme_project, user_initials="RSG", visible=True, port=port
+    )
+    return desktop
+
+
+# ------------------------------------ Initialize matrices -----------------------------------------
+
+# @_m.logbook_trace("Create and initialize matrices", save_arguments=True)
+def initialize_matrices(components, periods, scenario, delete_all_existing=False):
```

Comment:
is this initializing ALL matrices - or just transit skims? if former, then it doesn't belong here. If latter, then document and make obvious in name.